confirm.c

```diff
@@ -763,16 +765,18 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
 {
 	struct confirm_query *query;
-
-	query = malloc(sizeof(struct confirm_query));
+	query = malloc(sizeof(*query));
 	if(!query) logea(__FILE__, __LINE__, NULL);
 
+	if(flowid > CONFIRM_MAX_FLOWID) {
+		logd(LOG_WARN, "%s,%d: flowid > 127!\n", __FILE__, __LINE__);
```

Comment:
For ICMP the flowid also limits the number of probes sent at once, it is passed in the confirm_data_pack together with the ttl, so it is used to match the response too, right?

I did the same for TCP, I tried to keep the same variable (flowid) for matching, but I think it is not the best way indeed, since in TCP we have much more room for parallel probes, and don't need to be limited by MAX_FLOWID. That's why I think we should have another field in the confirm_query struct, like probe_number to control the concurrent probes and to match response, and stop using flowid to match response, at least for TCP. Is it ok to do so?

For the Linode parallel version I added another field in the confirm_query struct to control the instance number, the number goes in the first 16 bits of the sequence number, not in this pull request. If we add the probe_number field, it will solve this too, mda6 can use the field to control the parallel instances.